extensions/default/src/Toolbar/ToolbarLayoutSelector.tsx

```diff
@@ -50,43 +50,19 @@ function LayoutSelector({
     };
   }, [isOpen]);
 
-  useEffect(() => {
-    /* Reset to default layout when component unmounts */
-    return () => {
-      viewportGridService.setLayout({ numCols: 1, numRows: 1 });
-    };
-  }, []);
-
   const onInteractionHandler = () => setIsOpen(!isOpen);
   const DropdownContent = isOpen ? OHIFLayoutSelector : null;
 
-  const onSelectionHandler = ({ numRows, numCols }) => {
-    // TODO Introduce a service to persist the state of the current hanging protocol/app.
-
-    // TODO Here the layout change will amount to a change of hanging protocol as specified by the extension for this layout selector tool
-    // followed by the change of the grid itself.
-    if (hangingProtocolService.getActiveProtocol().protocol.id === 'mpr') {
-      toolbarService.recordInteraction({
-        groupId: 'MPR',
-        itemId: 'MPR',
-        interactionType: 'toggle',
-        commands: [
-          {
-            commandName: 'toggleMPR',
-            commandOptions: {},
-            context: 'CORNERSTONE',
-          },
-        ],
-      });
-    }
-
-    // When a new layout is selected, keep any extra/offscreen viewports
-    // so that if any of those viewports were populated via the UI then they
-    // will be maintained in case those viewports are redisplayed later.
-    viewportGridService.setLayout({
-      numRows,
-      numCols,
-      keepExtraViewports: true,
+  const onSelectionHandler = props => {
+    toolbarService.recordInteraction({
+      interactionType: 'action',
+      commands: [
+        {
+          commandName: 'setViewportLayout',
```

Comment:
I guess you can remove the `viewportGridService` in the top as well along with `use...`